module/apmot/span_test.go

```diff
@@ -0,0 +1,58 @@
+package apmot
+
+import (
+	"go.elastic.co/apm"
+	"testing"
+)
+
+func BenchmarkSetSpanContext(b *testing.B) {
```

Comment:
I think these benchmarks are a bit too low level, seeing as it's only possible for `setSpanContext` and `setTransactionContext` to be called via `Span.Finish`/`FinishWithOptions`.

How about changing them to start and finish real OT spans? i.e.

```go
tracer := apmot.New(apmot.WithTracer(apmtest.DiscardTracer))
for i := 0; i < b.N; i++ {
    span := tracer.StartSpanWithOptions("span", opentracing.StartSpanOptions{
        Tags: map[string]interface{}{/* tags go here */},
    })
    span.Finish()
}
```

Reply:
Unfortunately, `tracer` in this case is of `opentracing.Tracer` type, which doesn't have `StartSpanWithOptions()`, I may have to go with `StartSpan()`. Will look into it tomorrow.

Reply:
Sorry, I was conflating the implementation with the interface. You could instead pass an `opentracing.Tags` into `tracer.StartSpan`.

Reply:
Benchmark span finish.
Before
```
$ go test -bench=. tracer_test.go
goos: linux
goarch: amd64
pkg: go.elastic.co/apm/module/apmot
BenchmarkSpanSetSpanContext-4   	  200000	     10252 ns/op
BenchmarkSpanSetTxContext-4     	  100000	     11307 ns/op
PASS
ok  	go.elastic.co/apm/module/apmot	3.552s
```
After
```
$ go test -bench=. tracer_test.go
goos: linux
goarch: amd64
pkg: go.elastic.co/apm/module/apmot
BenchmarkSpanSetSpanContext-4   	  200000	      7656 ns/op
BenchmarkSpanSetTxContext-4     	  200000	      8064 ns/op
PASS
ok  	go.elastic.co/apm/module/apmot	3.480s
```

Reply:
Nice, thanks! Will you delete span_test.go? I think the benchmarks you added to tracer_test.go are sufficient?

Reply:
done.